src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -317,6 +340,11 @@ abstract Builder setDisableCertificateValidation(
      */
     public Builder withUrl(ValueProvider<String> url) {
       checkArgument(url != null, "withURL(url) called with null input.");
+      if (url.isAccessible()) {
+        checkArgument(
+            isValidUrlFormat(url.get()),
+            "Invalid url format. Url format should match [PROTOCOL]://[HOST]:[PORT]");
```

Comment:
Removed [] brackets and defined an `INVALID_URL_FORMAT_MESSAGE` constant with the additional error details.

Reply:
Added angular brackets around PORT since it would be optional now. So the error message looks like `Invalid url format. Url format should match PROTOCOL://HOST[:PORT], where PORT is optional. Supported Protocols are http and https. eg: http://hostname:8088`